apps/condo/domains/common/components/GraphQlSearchInputWithCheckAll.tsx

```diff
@@ -0,0 +1,91 @@
+import { Col, Form, FormInstance, FormItemProps, Row } from 'antd'
+import { isFunction } from 'lodash'
+
+import React, { ComponentProps, useCallback, useEffect, useState } from 'react'
+import { useIntl } from '@condo/next/intl'
+import { useValidations } from '@condo/domains/common/hooks/useValidations'
+
+import Checkbox from './antd/Checkbox'
+import { GraphQlSearchInput } from './GraphQlSearchInput'
+
+type InputWithCheckAllProps = {
+    onCheckBoxChange?: (value: boolean) => void
+    selectFormItemProps?: FormItemProps
+    checkAllFieldName: string,
+    checkAllInitialValue: boolean,
+    CheckAllMessage: string,
+    selectProps: ComponentProps<typeof GraphQlSearchInput>
+    checkBoxOffset?: number
+    form: FormInstance
+}
+
+export const GraphQlSearchInputWithCheckAll: React.FC<InputWithCheckAllProps> = (
+    {
+        onCheckBoxChange,
+        selectFormItemProps,
+        checkAllFieldName,
+        checkAllInitialValue,
+        selectProps,
+        CheckAllMessage,
+        form,
+        checkBoxOffset,
+    }
+) => {
+    const intl = useIntl()
+    const CheckedAllMessage = intl.formatMessage({ id: 'CheckedAll' })
+
+    const [isAllChecked, setIsAllChecked] = useState<boolean>(checkAllInitialValue)
+    const { requiredValidator } = useValidations()
+
+    const handleCheckboxChange = useCallback((event) => {
+        const value = event.target.checked
+        setIsAllChecked(value)
+
+        if (isFunction(onCheckBoxChange)) {
+            onCheckBoxChange(value)
+        }
+    }, [onCheckBoxChange])
+
+    const formItemName = String(selectFormItemProps.name)
+    useEffect(() => {
+        if (isAllChecked) {
+            form.setFieldsValue({ [formItemName]: [] })
+        }
+    }, [form, formItemName, isAllChecked])
+
+    const rules = []
```

Comment:
This will be recreated on each render tick. What the point storing data this way?